packages/edit-post/src/components/start-page-options/index.js

```diff
@@ -0,0 +1,121 @@
+/**
+ * WordPress dependencies
+ */
+import { Modal } from '@wordpress/components';
+import { __ } from '@wordpress/i18n';
+import { useState, useEffect } from '@wordpress/element';
+import {
+	store as blockEditorStore,
+	__experimentalBlockPatternsList as BlockPatternsList,
+} from '@wordpress/block-editor';
+import { useSelect, useDispatch } from '@wordpress/data';
+import { useAsyncList } from '@wordpress/compose';
+import { store as editorStore } from '@wordpress/editor';
+
+/**
+ * Internal dependencies
+ */
+import { store as editPostStore } from '../../store';
+
+function PatternSelection( { onChoosePattern } ) {
+	const { blockPatterns } = useSelect( ( select ) => {
+		const { __experimentalGetPatternsByBlockTypes } = select(
+			blockEditorStore
+		);
+		return {
+			blockPatterns: __experimentalGetPatternsByBlockTypes(
+				'core/post-content'
+			),
+		};
+	}, [] );
+	const shownBlockPatterns = useAsyncList( blockPatterns );
+	const { resetEditorBlocks } = useDispatch( editorStore );
+	useEffect( () => {
+		if ( blockPatterns.length <= 1 ) {
+			onChoosePattern();
+		}
+	}, [ blockPatterns.length ] );
+	return (
+		<BlockPatternsList
+			blockPatterns={ blockPatterns }
+			shownPatterns={ shownBlockPatterns }
+			onClickPattern={ ( _pattern, blocks ) => {
+				resetEditorBlocks( blocks );
+				onChoosePattern();
+			} }
+		/>
+	);
+}
+
+const START_PAGE_MODAL_STATES = {
+	INITIAL: 'INITIAL',
+	PATTERN: 'PATTERN',
+	CLOSED: 'CLOSED',
+};
+
+export default function StartPageOptions() {
+	const [ modalState, setModalState ] = useState(
+		START_PAGE_MODAL_STATES.INITIAL
+	);
+	const shouldOpenModel = useSelect(
+		( select ) => {
+			if ( modalState !== START_PAGE_MODAL_STATES.INITIAL ) {
```

Comment:
I got a bit confused with this.. What is the purpose of `INITIAL` state and its check here? The `shouldOpenModel` value isn't dependent on the state, so could we do that in a `useEffect` and have the `useSelect` with no dependencies?

Reply:
Hi @ntsekouras,
If the user already closed the modal we are in state CLOSED should open model should be false or if the model is already open shouldOpenModel should also be false. We could do these useEffect and have the useSelect with no dependencies but in that case, even if the user already close the model and is typing the useSelect would always be calling getCurrentPostType, getEditedPostContent etc. The selector getEditedPostContent is particularly expensive as it requires serializing all the blocks and it changes on every keypress. So doing the check as we are doing makes sure no more selectors are called after the user closes the modal, it is a performance enhancement.

Reply:
Would anything change if we used a lighter selector like `getBlockCount`? No strong opinions on this, but just trying understand the decisions better 😄

Reply:
getBlockCount is not as reliable, if one focuses on the empty paragraph at the start of a page the block count is 1, but the edited post content is still empty and there is no content there yet.
![image](https://user-images.githubusercontent.com/11271197/165970907-bc6ab2ab-f38c-4773-919c-6898e856428d.png)

So getEditedPostContent is a better fit for this use case, but I would say even if we used getBlockCount it would be better to not be calling a selector all the time after the user closed the modal.